ingestion/src/metadata/ingestion/source/databricks.py

```diff
@@ -19,6 +26,69 @@
 from metadata.ingestion.source.sql_source_common import SQLConnectionConfig
 
 
+class STRUCT(String):
+    """The SQL STRUCT type."""
+
+    __visit_name__ = "STRUCT"
+
+
+class ARRAY(String):
+    """The SQL ARRAY type."""
+
+    __visit_name__ = "ARRAY"
+
+
+class MAP(String):
+    """The SQL MAP type."""
+
+    __visit_name__ = "MAP"
+
+
+_type_map.update({"struct": STRUCT, "array": ARRAY, "map": MAP})
+
+
+@reflection.cache
+def get_columns(self, connection, table_name, schema=None, **kw):
+    # override to get columns properly; the reason is because databricks
+    # presents the partition information differently from oss hive
+    rows = self._get_table_columns(connection, table_name, schema)
+    # Strip whitespace
+    rows = [[col.strip() if col else None for col in row] for row in rows]
+    # Filter out empty rows and comment
+    rows = [row for row in rows if row[0] and row[0] != "# col_name"]
+    result = []
+    for (col_name, col_type, _comment) in rows:
+        # Handle both oss hive and Databricks' hive partition header, respectively
+        if col_name in ("# Partition Information", "# Partitioning"):
+            break
+        # Take out the more detailed type information
+        # e.g. 'map<ixnt,int>' -> 'map'
+        #      'decimal(10,1)' -> decimal
+        raw_data_type = col_type
+        col_type = re.search(r"^\w+", col_type).group(0)
+        try:
+            coltype = _type_map[col_type]
+        except KeyError:
+            util.warn(
+                "Did not recognize type '%s' of column '%s'" % (col_type, col_name)
+            )
+            coltype = types.NullType
+
+        col_info = {
+            "name": col_name,
+            "type": coltype,
+            "nullable": True,
+            "default": None,
+        }
+        if col_type in ["array", "struct", "map"]:
```

Comment:
`in` checks perform better on sets `{}` than on tuples and lists. Let's try to use those